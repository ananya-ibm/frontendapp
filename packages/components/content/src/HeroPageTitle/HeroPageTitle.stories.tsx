/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import { Button } from '@exo/frontend-components-base';
import React from 'react';
import { HeroPageTitle } from './HeroPageTitle';

export default {
  title: 'Components/Content/HeroPageTitle',
  component: HeroPageTitle
};

export const Default = args => <HeroPageTitle {...args} />;
Default.args = {
  title: 'Page title'
};

export const WithCustomContent = args => (
  <HeroPageTitle {...args}>
    <div style={{ display: 'inline-flex', alignItems: 'center' }}>
      <h2 style={{ marginRight: '1rem' }}>Custom content with a</h2> <Button label="button" />
    </div>
  </HeroPageTitle>
);
WithCustomContent.args = { ...Default.args };

export const WithImage = args => <HeroPageTitle {...args} />;
WithImage.args = {
  ...Default.args,
  image: '/static/automotive/batmobile.png'
};

export const WithColor = args => <HeroPageTitle {...args} />;
WithColor.args = {
  ...Default.args,
  background: 'red',
  foreground: 'white'
};
