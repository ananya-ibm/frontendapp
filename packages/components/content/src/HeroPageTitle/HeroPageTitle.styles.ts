/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import styled, { css } from 'styled-components';
import { ifProp, responsiveFontBlock } from '@exo/frontend-common-style-utils';
import theme from './HeroPageTitle.theme';

type HeroPageTitleProps = { image?: string; foreground?: string; background?: string };
export const HeroPageTitle = styled('div')<HeroPageTitleProps>`
  align-items: center;
  background-color: ${props => theme(props).backgroundColour};
  color: ${props => props.foreground ?? theme(props).textColour};
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 4.75rem;
  width: 100%;

  ${props => ifProp(props, 'image').then(css`
    background: url(${props.image}) no-repeat center;
    background-size: cover;
    color: ${props.theme.colors.inverse.base.fg};
  `)}

  ${props => ifProp(props, 'background').then(css`
    background-color: ${props.background};
  `)}

  ${props => ifProp(props, 'foreground').then(css`
    color: ${props.foreground};
  `)}
`;

export const Title = styled('h1')`
  ${props => responsiveFontBlock(props.theme.typography.display.heading1)};
  letter-spacing: 0.07rem;
  text-align: center;
`;
