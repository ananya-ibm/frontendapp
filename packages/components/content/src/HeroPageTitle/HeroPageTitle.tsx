/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import * as S from './HeroPageTitle.styles';

export const HeroPageTitle = ({ image, background, foreground, children, title }: Props) => {
  return (
    <S.HeroPageTitle image={image} background={background} foreground={foreground}>
      {title && <S.Title>{title}</S.Title>}
      {children}
    </S.HeroPageTitle>
  );
};

type Props = {
  children?: any;
  background?: string;
  foreground?: string;
  image?: string;
  title?: string;
};
