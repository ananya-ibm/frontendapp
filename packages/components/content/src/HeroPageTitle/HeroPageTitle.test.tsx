/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { renderWithTheme as render } from '@testUtils';

import { axe } from 'jest-axe';
import { HeroPageTitle } from './HeroPageTitle';

const testProps = {
  title: 'A title',
  children: 'some children',
  color: 'blue',
  image: 'image.jpg'
};

describe('<HeroPageTitle /> component', () => {
  test('its snapshot matches', () => {
    const { container } = render(<HeroPageTitle {...testProps} />);

    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();
  });
  test('it passes an axe accesibility check', async () => {
    const { container } = render(<HeroPageTitle {...testProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
