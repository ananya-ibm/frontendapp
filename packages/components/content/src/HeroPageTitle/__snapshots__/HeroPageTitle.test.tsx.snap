// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroPageTitle /> component its snapshot matches 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f4f4f4;
  color: #161616;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 4.75rem;
  width: 100%;
  background: url(image.jpg) no-repeat center;
  background-size: cover;
  color: #ffffff;
}

.c1 {
  font: 600 2.625rem/3.12375rem 'IBM Plex Sans',Arial,sans-serif;
  -webkit-letter-spacing: 0.07rem;
  -moz-letter-spacing: 0.07rem;
  -ms-letter-spacing: 0.07rem;
  letter-spacing: 0.07rem;
  text-align: center;
}

@media (min-width:672px) {
  .c1 {
    line-height: 4.887499999999999rem;
    font-size: 4.25rem;
  }
}

@media (min-width:1056px) {
  .c1 {
    line-height: 6.3825rem;
    font-size: 5.75rem;
    -webkit-letter-spacing: -0.64px;
    -moz-letter-spacing: -0.64px;
    -ms-letter-spacing: -0.64px;
    letter-spacing: -0.64px;
  }
}

@media (min-width:1312px) {
  .c1 {
    line-height: 8.158750000000001rem;
    font-size: 7.625rem;
    -webkit-letter-spacing: -0.64px;
    -moz-letter-spacing: -0.64px;
    -ms-letter-spacing: -0.64px;
    letter-spacing: -0.64px;
  }
}

@media (min-width:1584px) {
  .c1 {
    line-height: 10.2375rem;
    font-size: 9.75rem;
    -webkit-letter-spacing: -0.96px;
    -moz-letter-spacing: -0.96px;
    -ms-letter-spacing: -0.96px;
    letter-spacing: -0.96px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    A title
  </h1>
  some children
</div>
`;
